Yoast/wordpress-seo
# Add Indexable REST API endpoints

## Summary

This PR can be summarized in the following changelog entry:

* Adds the `PATCH` endpoint for indexables.

## Relevant technical choices:

* Both `POST` and `PUT` have been omitted as they don't have any use for indexables. This is due to a combination of many optional fields and that an indexable cannot exist as a standalone object.

## Test instructions

This PR can be tested by following these steps:

**Before testing**
* Ensure you have a tool such as Postman installed on your machine.
* Within the Authorization section of Postman, select Basic Auth and fill in your WordPress admin username and password.
* Ensure you have https://github.com/WP-API/Basic-Auth installed in your testing environment.

**Testing**
* Ensure the pre-existing `GET` endpoint still works. You can do this by entering the following call in Postman: `local.wordpress.test/wp-json/yoast/v1/indexables/<id of post>?object_type=post`. Please replace `<id of post>` with a valid post ID.
* Ensure you are creating a `GET` request and click Send. (See screenshot).

<img width="1351" alt="postman" src="https://user-images.githubusercontent.com/4181340/42444101-93d1d2ea-836f-11e8-86e8-1f6c9b8f3345.png">

* See that you are presented with a JSON-encoded response.
* Do the same request again, but for a term, by altering `object_type`to `term` and looking up a valid term ID (i.e. the ID of a particular category).

* To test the `PATCH` endpoint, you need to edit the type of request. (See screenshot).

<img width="1361" alt="postman" src="https://user-images.githubusercontent.com/4181340/42447703-14dc3dd4-837b-11e8-971a-2c1fe03ce8e9.png">

* Add a few fields (note that you are not required to send all of them). Use the screenshot as an example.
* Also add a few fields with no value. Use these to determine that they aren't processed.
* Click send and see that you receive the message "Patch successful".
* Do the same request again, but for a term, by altering `object_type`to `term` and looking up a valid term ID (i.e. the ID of a particular category).

## Quality assurance


Fixes #9519
